api/uniter/unit_test.go

```diff
@@ -684,6 +684,86 @@ func (s *unitSuite) TestWatchActionNotificationsMoreResults(c *gc.C) {
 	c.Assert(err.Error(), gc.Equals, "expected 1 result, got 2")
 }
 
+func (s *unitSuite) TestWatchUpgradeSeriesNotifications(c *gc.C) {
+	watcher, err := s.apiUnit.WatchUpgradeSeriesNotifications()
+	c.Assert(err, jc.ErrorIsNil)
+
+	notifyWatcher := watchertest.NewNotifyWatcherC(c, watcher, s.BackingState.StartSync)
+	defer notifyWatcher.AssertStops()
+
+	// Why do I have two initial events?!?
+	_, ok := <-notifyWatcher.Watcher.Changes()
+	notifyWatcher.Assert(ok, jc.IsTrue)
+
+	notifyWatcher.AssertOneChange()
+
+	s.CreateUpgradeSeriesLock(c)
+
+	// Expect a notification that the document was created (i.e. a lock was placed)
+	notifyWatcher.AssertOneChange()
+
+	err = s.wordpressMachine.RemoveUpgradeSeriesLock()
+	c.Assert(err, jc.ErrorIsNil)
+
+	// A notification that the document was removed (i.e. the lock was released)
+	notifyWatcher.AssertOneChange()
+}
+
+func (s *unitSuite) TestUpgradeSeriesStatus(c *gc.C) {
+	// First we create the prepare lock
```

Comment:
Is it worth checking the pre-condition of s.apiUnit.UpgradeSeriesStatus() before we create the lock? So that we see the value changes to Started?

It seems odd that you have this test, which says the status is Started, even though there isn't a call to SetUpgradeSeriesStatus(UnitStarted).
Shouldn't the status of a *unit* start in something like Pending?

If we want to just have "started" as the initial state, then I'd say we don't need a test that you can set the state to Started (drop TestSetUpgradeSeriesStatus, or change it to set the status to Completed)
or clarify that we just treat setting the status to Started as a no-op?
Do we support going from Completed back to Started? Or should that be considered an error? (I'd tend toward the latter, though re-running a pre upgrade step is something we've discussed)

Reply:
UpgradeSeriesStatus only makes a call to the api to get the remote state regardless of what it is. When a lock is created the status just so happens to be started, since the act of creating a lock or `juju upgrade-series prepare...` starts the upgrade process.

Reply:
Yes, but correct, lets test our assumption!